pkg/kubelet/rkt/rkt.go

```diff
@@ -1108,8 +1110,42 @@ func (r *Runtime) RunInContainer(containerID string, cmd []string) ([]byte, erro
 	args := append([]string{}, "enter", fmt.Sprintf("--app=%s", id.appName), id.uuid)
 	args = append(args, cmd...)
 
-	result, err := r.runCommand(args...)
-	return []byte(strings.Join(result, "\n")), err
+	result, err := r.buildCommand(args...).CombinedOutput()
+	if err != nil {
+		if exitErr, ok := err.(*exec.ExitError); ok {
+			err = &rktExitError{exitErr}
+		}
+	}
+	return result, err
+}
+
+var _ utilExec.ExitError = &rktExitError{}
+
+// rktExitError implemets /pkg/util/exec.ExitError interface.
+type rktExitError struct {
```

Comment:
Generally I agree with you, but in the same file where that type is defined I think it can be treated as more of a "constructor" as opposed to assuming implementation details. That said, I'm still new to Go, so let me know if there's something I'm missing here.